src/command/client/search/interactive.rs

```diff
@@ -170,107 +170,43 @@ impl State {
     }
 
     #[allow(clippy::cast_possible_truncation)]
-    fn draw<T: Backend>(&mut self, f: &mut Frame<'_, T>, results: &[History]) {
-        let chunks = Layout::default()
-            .direction(Direction::Vertical)
-            .margin(0)
-            .constraints([
-                Constraint::Length(3),
-                Constraint::Min(1),
-                Constraint::Length(3),
-            ])
-            .split(f.size());
-
-        let top_chunks = Layout::default()
-            .direction(Direction::Horizontal)
-            .constraints([Constraint::Percentage(50); 2])
-            .split(chunks[0]);
-
-        let top_left_chunks = Layout::default()
-            .direction(Direction::Vertical)
-            .constraints([Constraint::Length(1); 3])
-            .split(top_chunks[0]);
-
-        let top_right_chunks = Layout::default()
-            .direction(Direction::Vertical)
-            .constraints([Constraint::Length(1); 3])
-            .split(top_chunks[1]);
-
-        let title = if self.update_needed.is_some() {
-            let version = self.update_needed.clone().unwrap();
-
-            Paragraph::new(Text::from(Span::styled(
-                format!(" Atuin v{VERSION} - UPDATE AVAILABLE {version}"),
-                Style::default().add_modifier(Modifier::BOLD).fg(Color::Red),
-            )))
+    #[allow(clippy::bool_to_int_with_if)]
+    fn draw<T: Backend>(
+        &mut self,
+        f: &mut Frame<'_, T>,
+        results: &[History],
+        compact: bool,
+        show_preview: bool,
+    ) {
+        let border_size = if compact { 0 } else { 1 };
+        let preview_width = f.size().width - 2;
+        let preview_height = if show_preview {
+            let longest_command = results
+                .iter()
+                .max_by(|h1, h2| h1.command.len().cmp(&h2.command.len()));
+            longest_command.map_or(0, |v| {
+                std::cmp::min(
+                    4,
+                    (v.command.len() as u16 + preview_width - 1 - border_size)
```

Comment:
Just noticed a regression following the merge of normal and compact modes, where commands with more than 2 lines, only show 2 lines, while the max should be 4 lines. This affects both modes.

Reply:
Issue no longer reproduces.